### PR TITLE
Add /api/redriect/recommend route to force session creation with given attributed_to value

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Alternative approach to setting attributed_to - make a special API route to force session creation

## Justify why they are needed

More stable solution than relying on query param when starting the app (https://github.com/HedvigInsurance/racoon/pull/2151?no-redirect=1) 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2159** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2159" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2159?utm_source=stack-comment).

File: apps/store/src/pages/api/redirect/recommend.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { setupShopSessionServiceServerSide } from '@/services/shopSession/ShopSession.helpers'
+import { getCountryByLocale } from '@/utils/l10n/countryUtils'
+import { getUrlLocale } from '@/utils/l10n/localeUtils'
+import { ORIGIN_URL, PageLink } from '@/utils/PageLink'
+
+const recommendHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { query = {} } = req
+  const { attributed_to: attributedTo, next = '' } = query
+
+  if (!attributedTo || Array.isArray(attributedTo)) {
```
Comment (robinandeer) [inline apps/store/src/pages/api/redirect/recommend.ts:12]: I would just do:

```
if (typeof attributedTo !== 'string') {
```

File: apps/store/src/pages/api/redirect/recommend.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    return res.status(400).send('Expected attributed_to query param with single value')
+  }
+  if (Array.isArray(next)) {
```
Comment (robinandeer) [inline apps/store/src/pages/api/redirect/recommend.ts:15]: same here probably

File: apps/store/src/pages/api/redirect/recommend.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    return res.status(400).send('Multiple next values not supported')
+  }
+
+  const routingLocale = getUrlLocale(next)
+  const { countryCode } = getCountryByLocale(routingLocale)
+
+  const apolloClient = await initializeApolloServerSide({ req, res })
+  const shopSessionService = setupShopSessionServiceServerSide({ apolloClient, req, res })
+  const shopSession = await shopSessionService.create({ countryCode, attributedTo })
+  shopSessionService.saveId(shopSession.id)
```
Comment (robinandeer) [inline apps/store/src/pages/api/redirect/recommend.ts:25]: We might want to try/catch here so we make sure to redirect the user somewhere meaningful even if they encounter an error

**Review by robinandeer — APPROVED** [review]
Neat!

File: apps/store/src/pages/api/redirect/recommend.ts (hunk 1 above)
Comment (alebedev) [reply apps/store/src/pages/api/redirect/recommend.ts:12]: Good idea.  There's also empty string, but let's ignore it

**Pull request merged**